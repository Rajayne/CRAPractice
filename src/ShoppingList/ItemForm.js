import React, { useState } from "react";

const NewItemForm = ({ addItem }) => {
  const initialState = {
    name: "",
    qty: "",
  };
  const [formData, setFormData] = useState(initialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({ ...formData, [name]: value }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    addItem({ ...formData });
    setFormData(initialState);
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="name">Product Name: </label>
      <input
        type="text"
        id="name"
        name="name"
        placeholder="Item Name"
        value={formData.name}
        onChange={handleChange}
      ></input>

      <label htmlFor="qty">Quantity: </label>
      <input
        type="text"
        id="qty"
        name="qty"
        placeholder="Quantity"
        value={formData.qty}
        onChange={handleChange}
      ></input>

      <button>Add Item</button>
    </form>
  );
};

export default NewItemForm;
