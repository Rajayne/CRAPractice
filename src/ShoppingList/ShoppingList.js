import React, { useState } from "react";
import Item from "./Item";
import NewItemForm from "./ItemForm";

const ShoppingList = () => {
  const initialState = [
    { id: 1, name: "Peanut Butter", qty: 2 },
    { id: 2, name: "Lactaid Milk", qty: 1 },
  ];
  const [items, setItems] = useState(initialState);

  const addItem = (name, qty) => {
    setItems((items) => [...items, { name, qty }]);
  };

  return (
    <div>
      <h3>Shopping List</h3>
      <NewItemForm addItem={addItem} />
      {items.map(({ id, name, qty }) => (
        <Item id={id} name={name} qty={qty} key={id} />
      ))}
    </div>
  );
};

export default ShoppingList;
