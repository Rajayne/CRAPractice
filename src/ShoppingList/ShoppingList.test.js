import React from "react";
import { render, fireEvent, getByLabelText } from "@testing-library/react";
import ShoppingList from "./ShoppingList";

test("renders without crashing", function () {
  render(<ShoppingList />);
});

test("matches snapshot", function () {
  const { asFragment } = render(<ShoppingList />);
  expect(asFragment()).toMatchSnapshot();
});

test("should add new item", function () {
  const { queryByText, getByLabelText } = render(<ShoppingList />);
  const input = getByLabelText("Product:");
  const btn = queryByText("Add Item");
  expect(queryByText("Product Name: Chocolate Milk")).not.toBeInTheDocument();
  fireEvent.change(input, { target: { value: "Chocolate Milk" } });
  fireEvent.click(btn);
  expect(queryByText("Product Name: Chocolate Milk")).toBeInTheDocument();
});
