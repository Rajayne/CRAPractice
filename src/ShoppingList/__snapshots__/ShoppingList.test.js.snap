// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Shopping List
    </h3>
    <form>
      <label
        for="name"
      >
        Product: 
      </label>
      <input
        id="name"
        name="name"
        placeholder="Item Name"
        type="text"
        value=""
      />
      <label
        for="qty"
      >
        Quantity: 
      </label>
      <input
        id="qty"
        name="qty"
        placeholder="Quantity"
        type="text"
        value=""
      />
      <button>
        Add Item
      </button>
    </form>
    <ul>
      <li>
        Product Name: Peanut Butter
      </li>
      <li>
        Quantity: 2
      </li>
    </ul>
    <ul>
      <li>
        Product Name: Lactaid Milk
      </li>
      <li>
        Quantity: 1
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
